src/main/java/ch/njol/skript/effects/EffLog.java

```diff
@@ -46,28 +46,29 @@
 import ch.njol.util.Closeable;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Log")
 @Description({"Writes text into a .log file. Skript will write these files to /plugins/Skript/logs.",
 		"NB: Using 'server.log' as the log file will write to the default server log. Omitting the log file altogether will log the message as '[Skript] [&lt;script&gt;.sk] &lt;message&gt;' in the server log."})
 @Examples({"on place of TNT:",
-		"	log \"%player% placed TNT in %world% at %location of block%\" to \"tnt/placement.log\""})
-@Since("2.0")
+		"   log \"%player% placed TNT in %world% at %location of block%\"",
+		"	log \"%player% placed TNT in %world% at %location of block%\" to file \"tnt/placement.log\"",
+		"	log \"%player% placed TNT in %world% at %location of block%\" to file \"tnt/placement.log\"\"with a severity of warning\""})
+
+
+@Since("2.0, INSERT VERSION (Adds severity to logs)")
 public class EffLog extends Effect {
 	static {
-		Skript.registerEffect(EffLog.class, "log %strings% [(to|in) [file[s]] %-strings%]");
+		Skript.registerEffect(EffLog.class, "log %strings% [(to|in) [file[s]] %-strings%] [with [the|a] severity [of] (1:warning|2:severe)]");
 	}
 	
-	private final static File logsFolder = new File(Skript.getInstance().getDataFolder(), "logs");
+	private static File logsFolder = new File(Skript.getInstance().getDataFolder(), "logs");
 	
-	final static HashMap<String, PrintWriter> writers = new HashMap<>();
+	static HashMap<String, PrintWriter> writers = new HashMap<>();
 	static {
 		Skript.closeOnDisable(new Closeable() {
 			@Override
 			public void close() {
-				for (final PrintWriter pw : writers.values())
+				for (PrintWriter pw : writers.values())
```

Comment:
While it probably saves some resources not closing and opening PrintWriter instances everytime this Effect is called, I don't think these instances should just all be closed when Skript disables as its generally a good idea to close resources as soon as you’re done using them to prevent resource leaks

This is just a suggestion though.

Reply:
It's probably out of scope of the PR too